src/test/scheduler_tests.cpp

```diff
@@ -155,4 +155,45 @@ BOOST_AUTO_TEST_CASE(singlethreadedscheduler_ordered)
     BOOST_CHECK_EQUAL(counter2, 100);
 }
 
+BOOST_AUTO_TEST_CASE(mockforward)
+{
+    CScheduler scheduler;
+
+    int counter{0};
+    CScheduler::Function dummy = [&counter](){counter++;};
```

Comment:
nit, you can drop `()`.

Reply:
I think this is a style preference (plus it's in tests) so it's whatever.

For whatever reason, I prefer this form when it's a function that should be called multiple times (as dummy is) and the no () form when it's more akin to a std::bind for a call-once thing.

Reply:
updated